pkg/virt-operator/install-strategy/delete.go

```diff
@@ -38,6 +39,43 @@ import (
 	"kubevirt.io/kubevirt/pkg/virt-operator/util"
 )
 
+func deleteDummyWebhookValidators(kv *v1.KubeVirt,
+	clientset kubecli.KubevirtClient,
+	stores util.Stores,
+	expectations *util.Expectations) error {
+
+	kvkey, err := controller.KeyFunc(kv)
+	if err != nil {
+		return err
+	}
+
+	gracePeriod := int64(0)
+	deleteOptions := &metav1.DeleteOptions{
+		GracePeriodSeconds: &gracePeriod,
+	}
+
+	objects := stores.ValidationWebhookCache.List()
+	for _, obj := range objects {
+		if webhook, ok := obj.(*admissionregistrationv1beta1.ValidatingWebhookConfiguration); ok {
+
+			if webhook.DeletionTimestamp != nil {
+				continue
+			}
+			if key, err := controller.KeyFunc(webhook); err == nil {
+				expectations.ValidationWebhook.AddExpectedDeletion(kvkey, key)
+				err = clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Delete(webhook.Name, deleteOptions)
+				if err != nil {
+					expectations.ValidationWebhook.DeletionObserved(kvkey, key)
+					return fmt.Errorf("unable to delete validation webhook: %v", err)
```

Comment:
It would make sense to collect errors and keep going to the end of the list, would it not? This "return err" happens right in the middle of a for loop.

Reply:
that's the pattern everywhere for create and delete. The logic goes as far as it can until it hits an error, then it backs out and re-enqueues the key to try again. hopefully getting further the next time if the error is transient.